vowpalwabbit/memory.h

```diff
@@ -17,7 +19,11 @@ T* calloc_or_throw(size_t nmemb)
     const char* msg = "internal error: memory allocation failed!\n";
     // use low-level function since we're already out of memory.
     fputs(msg, stderr);
-    THROW(msg);
+	#ifdef VW_NOEXCEPT
```

Comment:
We could encapsulate this behavior in an allocator that is provided to VW that can the return null or throw.

Reply:
I think this is a good idea.  Don't want to attempt this as part of this PR.